src/org/traccar/database/DeviceManager.java

```diff
@@ -52,6 +55,13 @@
 
     private final Map<Long, Position> positions = new ConcurrentHashMap<>();
 
+    private final Map<Long, Boolean> motionStates = new ConcurrentHashMap<>();
+    private final Map<Long, Position> motionStatePositions = new ConcurrentHashMap<>();
```

Comment:
We need also distance for detect moving, link `positionId` to new event and pass appropriate `position` (whole) to notification.